api/src/Service/JWT/JWTService.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\JWT;
+
+use App\Entity\User;
+use Firebase\JWT\JWT;
+
+class JWTService
+{
+    public function __construct(private string $rootDir)
+    {
+    }
+
+    public function createNewJWT(User $user)
+    {
+        $privateKey = file_get_contents(sprintf('%s/config/jwt/private.pem', $this->rootDir));
+        $publicKey = file_get_contents(sprintf('%s/config/jwt/public.pem', $this->rootDir));
+
+        $kid = openssl_digest($publicKey, 'sha256');
+        $payload = [
+            'username' => $user->getEmail(),
+            'userId' => $user->getId(),
+            'aud' => 'registration_service',
+            'iat' => strtotime('now'),
+            'exp' => strtotime('+1 hour'),
+            'nbf' => strtotime('-10 seconds'),
+            'iss' => 'digideps',
+            'sub' => $user->getEmail(),
```

Comment:
Worth noting that jwt's don't encrypt, they just validate the payload, so email references might be considered a potential for data leakage if the token got exposed.

Reply:
True - I guess we could drop the two user fields and substitute the value of `sub` to be user ID and then grab from the DB where needed